graphql/query_resolver.go

```diff
@@ -978,13 +978,31 @@ func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions)
 
 		limit = limitOpt
 	}
+
 	requesters := options.Requesters
 	if len(requesters) == 0 {
 		requesters = evergreen.SystemVersionRequesterTypes
 	}
 
 	maxOrderOpt := utility.FromIntPtr(options.MaxOrder)
 	minOrderOpt := utility.FromIntPtr(options.MinOrder)
+	revision := utility.FromStringPtr(options.Revision)
+
+	if options.Revision != nil {
```

Comment:
Can we modularize this logic so it can also be used [here](https://github.com/evergreen-ci/evergreen/blob/ed6101e2f0ef85f8748910edc7de3065cabcb8f3/graphql/query_resolver.go#L814-L828)?